gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -1200,7 +1200,7 @@ async fn test_gateway_executes_swaps_between_connected_federations() -> anyhow::
                 "Gateway balance after swap".to_string(),
                 || async {
                     let post_balances = get_balances(&rpc, &[id1, id2]).await;
-                    if post_balances[0] == pre_balances[0] && post_balances[1] == pre_balances[1] {
+                    if post_balances[0] == pre_balances[0] || post_balances[1] == pre_balances[1] {
```

Comment:
I don't understand

Reply:
I should have linked this comment https://github.com/fedimint/fedimint/issues/3672#issuecomment-1824530248

Reply:
oh this is not the exit condition of retry but retry again condition.
makes a lot more sense now.